### PR TITLE
Improve handling for incorrect or missing DFE signin secret

### PR DESCRIPTION
**Pull request by TobyRet**
### Context

I ran into an issue running manage-courses-frontend locally for the first time. It is not known if this is a widespread issue but it may be worth adding some additional handling and troubleshooting guidance to the README for other new joiners.

[Trello ticket ](https://trello.com/c/D9Ym3r5I/2407-update-readme-manage-courses-frontend)

### Changes proposed in this pull request
- Improved error handling for incorrect or missing DFE sign in secret
- Update README with troubleshooting advice

### Checklist

- [x] Make sure all information from the Trello card is in here
- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

**Comment by misaka** [comment]
@TobyRet Could we add a `rescue` clause to try to provide a hint to the next poor soul that hits this?

https://github.com/DFE-Digital/manage-courses-frontend/blob/side-step-sign-in-for-development/config/initializers/omniauth.rb#L34-L43

**Comment by misaka** [comment]
Also, is there any way to detect that the assets haven't been precompiled, and to emit something more helpful?

**Comment by defong** [comment]
[Update README - manage-courses-frontend](https://trello.com/c/D9Ym3r5I/2407-update-readme-manage-courses-frontend)

**Comment by TobyRet** [comment]
> https://github.com/DFE-Digital/manage-courses-frontend/blob/side-step-sign-in-for-development/config/initializers/omniauth.rb#L34-L43

Yes was thinking about this last night @misaka - will make the change

File: config/initializers/omniauth.rb [hunk 1]
```diff
@@ -40,13 +40,22 @@ def callback_phase
...
     end
   end
 end
 
 class OmniAuth::Strategies::Dfe < OmniAuth::Strategies::OpenIDConnect; end
 
+# Removing the default secret from config/settings.yml allows app to fail fast
+# with explicit reason as to why
+if !Settings.dfe_signin.secret.present?
+  raise 'dfe_signin secret not set in config/settings'
```
Comment (defong) [inline config/initializers/omniauth.rb]: is this just for dev env?

Comment (TobyRet) [reply config/initializers/omniauth.rb]: It will in any environment currently ...

File: config/initializers/omniauth.rb [hunk 2]
```diff
@@ -40,13 +40,22 @@ def callback_phase
         fail!(:timeout, e)
       rescue ::SocketError => e
         fail!(:failed_to_connect, e)
+      # Adding this provides some nicer / additional logging but not why the error was caused
+      rescue Rack::OAuth2::Client::Error => e
+        fail!(:invalid_client, e)
```
Comment (misaka) [inline config/initializers/omniauth.rb]: Could we provide additional text saying something like:

> Could not perform OAuth authentication. Do you have 'dfe_signin.secret' set in Settings?

And we could expand on this as we find other common causes for this problem.

**Pull request closed without merging**